accounts/keystore/account_cache.go

```diff
@@ -0,0 +1,227 @@
+package keystore
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/accounts"
+	"github.com/0xPolygon/polygon-edge/helper/common"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+)
+
+// KeyStoreScheme is the protocol scheme prefixing account and wallet URLs.
+const KeyStoreScheme = "keystore"
+
+// accountCache is a live index of all accounts in the keystore.
+type accountCache struct {
+	logger   hclog.Logger
+	keyDir   string
+	mu       sync.Mutex
+	allMap   map[types.Address]encryptedKeyJSONV3
+	throttle *time.Timer
+	notify   chan struct{}
+}
+
+func newAccountCache(keyDir string, logger hclog.Logger) (*accountCache, chan struct{}) {
+	ac := &accountCache{
+		logger: logger,
+		keyDir: keyDir,
+		notify: make(chan struct{}, 1),
+		allMap: make(map[types.Address]encryptedKeyJSONV3),
+	}
+
+	if err := common.CreateDirSafe(keyDir, 0700); err != nil {
+		ac.logger.Info("can't create dir", "err", err)
+
+		return nil, nil
+	}
+
+	keysPath := path.Join(keyDir, "keys.txt")
+
+	ac.keyDir = keysPath
+
+	if _, err := os.Stat(keysPath); errors.Is(err, os.ErrNotExist) {
+		if _, err := os.Create(keysPath); err != nil {
+			ac.logger.Info("can't create new file", "err", err)
+
+			return nil, nil
+		}
+	}
+
+	ac.scanAccounts() //nolint:errcheck
+
+	return ac, ac.notify
+}
+
+func (ac *accountCache) accounts() []accounts.Account {
+	ac.mu.Lock()
+	defer ac.mu.Unlock()
+
+	cpy := make([]accounts.Account, len(ac.allMap))
+	i := 0
+
+	for addr := range ac.allMap {
+		cpy[i] = accounts.Account{Address: addr}
+		i++
```

Comment:
i++ is post increment, so it means it will increment a value after it is used in expression. Hence it is safe to use it like this:
```suggestion
		cpy[i++] = accounts.Account{Address: addr}
```

Reply:
This syntax doesn't work in golang